droidlet/interpreter/robot/tasks.py

```diff
@@ -620,3 +624,167 @@ def step(self):
             self.agent.mover.explore(self.goal)
             self.finished = self.agent.mover.nav.is_done_exploring().value
 
+class TimedExplore(TrajectorySaverTask):
+    """use slam to explore environemt"""
+
+    def __init__(self, agent, task_data):
+        super().__init__(agent, task_data)
+        self.command_sent = False
+        self.agent = agent
+        self.goal = task_data.get("goal")
+        self.timeout = task_data.get("timeout")
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    @Task.step_wrapper
+    def step(self):
+        super().step()
+        self.interrupted = False
+        self.finished = False
+        if not self.finished and self.timeout > 0:
+            self.agent.mover.explore(self.goal)
+            self.timeout -= 1
+            self.finished = self.agent.mover.nav.is_done_exploring().value
```

Comment:
maybe add a `self.mover.is_done_exploring()` and use that?

Reply:
yea, good idea.